homeassistant/components/recorder/db_schema.py

```diff
@@ -128,7 +128,7 @@ class LegacyBase(DeclarativeBase):
 LEGACY_MAX_LENGTH_EVENT_CONTEXT_ID: Final = 36
 CONTEXT_ID_BIN_MAX_LENGTH = 16
 
-MYSQL_COLLATE = "utf8mb4_unicode_ci"
+MYSQL_COLLATE = "utf8mb4_bin"
```

Comment:
Note to self: We need to check the minimum version of when this appeared against supported versions

Reply:
MySQL: Added in version 5.5 (https://downloads.mysql.com/docs/mysql-5.5-relnotes-en.pdf), our minimum version is currently 8.0.0
MariaDB: Added in version 5.5, our minimum version is currently 10.3.0